bin/reth/src/node/cl_events.rs

```diff
@@ -0,0 +1,86 @@
+//! Events related to Consensus Layer health.
+
+use futures::Stream;
+use reth_provider::CanonChainTracker;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+use tokio::time::Interval;
+
+/// Interval of checking Consensus Layer client health.
+const CHECK_INTERVAL: Duration = Duration::from_secs(300);
+/// Period of not exchanging transition configurations with Consensus Layer client,
+/// after which the warning is issued.
+const NO_TRANSITION_CONFIG_EXCHANGED_PERIOD: Duration = Duration::from_secs(120);
+/// Period of not receiving fork choice updates from Consensus Layer client,
+/// after which the warning is issued.
+const NO_FORKCHOICE_UPDATE_RECEIVED_PERIOD: Duration = Duration::from_secs(120);
+
+/// A Stream of [ConsensusLayerHealthEvent].
+pub struct ConsensusLayerHealthEvents {
+    interval: Interval,
+    canon_chain: Box<dyn CanonChainTracker>,
+}
+
+impl ConsensusLayerHealthEvents {
+    /// Creates a new [ConsensusLayerHealthEvents] with the given canonical chain tracker.
+    pub fn new(canon_chain: Box<dyn CanonChainTracker>) -> Self {
+        Self { interval: tokio::time::interval(CHECK_INTERVAL), canon_chain }
+    }
+}
+
+impl Stream for ConsensusLayerHealthEvents {
+    type Item = ConsensusLayerHealthEvent;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = self.get_mut();
+
+        if this.interval.poll_tick(cx).is_ready() {
+            // this ensures the interval will be polled periodically, see [Interval::poll_tick]
+            let _ = this.interval.poll_tick(cx);
```

Comment:
style nit: since only have one check inside this if, this can be a `while .is_ready()` or 
`loop{ready!()}`

Reply:
great, thank you